quickstart/201-azfw-multi-addresses/main.tf

```diff
@@ -0,0 +1,227 @@
+  admin_username        = var.admin_username
+  admin_password        = random_password.password[count.index].result
+  network_interface_ids = [azurerm_network_interface.backend_nic[count.index].id]
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+  source_image_reference {
+    publisher = "MicrosoftWindowsServer"
+    offer     = "WindowsServer"
+    sku       = "2019-Datacenter"
+    version   = "latest"
+  }
+}
+
+resource "azurerm_firewall_policy" "azfw_policy" {
+  name                     = "azfw-policy"
+  resource_group_name      = azurerm_resource_group.rg.name
+  location                 = azurerm_resource_group.rg.location
+  sku                      = var.firewall_sku_tier
+  threat_intelligence_mode = "Alert"
+}
+
+resource "azurerm_firewall_policy_rule_collection_group" "policy_rule_collection_group" {
+  name               = "RuleCollectionGroup"
+  firewall_policy_id = azurerm_firewall_policy.azfw_policy.id
+  priority           = 300
+  application_rule_collection {
+    name     = "web"
+    priority = 100
+    action   = "Allow"
+    rule {
+      name = "wan-address"
+      protocols {
+        type = "Http"
+        port = 80
+      }
+      protocols {
+        type = "Https"
+        port = 443
+      }
+      destination_fqdns = ["getmywanip.com"]
+      source_addresses  = ["*"]
+    }
+    rule {
+      name = "google"
+      protocols {
+        type = "Http"
+        port = 80
+      }
+      protocols {
+        type = "Https"
+        port = 443
+      }
+      destination_fqdns = ["www.google.com"]
+      source_addresses  = ["10.10.1.0/24"]
+    }
+    rule {
+      name = "wupdate"
+      protocols {
+        type = "Http"
+        port = 80
+      }
+      protocols {
+        type = "Https"
+        port = 443
+      }
+      destination_fqdn_tags = ["WindowsUpdate"]
+      source_addresses      = ["*"]
+    }
+  }
+  nat_rule_collection {
+    name     = "Coll-01"
+    action   = "Dnat"
+    priority = 200
+    rule {
+      name                = "rdp-01"
+      protocols           = ["TCP"]
+      translated_address  = "10.10.1.4"
+      translated_port     = "3389"
+      source_addresses    = ["*"]
+      destination_address = azurerm_public_ip.pip_azfw.ip_address
+      destination_ports   = ["3389"]
+    }
+    rule {
+      name                = "rdp-02"
+      protocols           = ["TCP"]
+      translated_address  = "10.10.1.5"
+      translated_port     = "3389"
+      source_addresses    = ["*"]
+      destination_address = azurerm_public_ip.pip_azfw.ip_address
+      destination_ports   = ["3389"]
+    }
+  }
+}
+
+resource "azurerm_firewall" "fw" {
+  name                = "azfw"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  sku_name            = "AZFW_VNet"
+  sku_tier            = var.firewall_sku_tier
+  ip_configuration {
```

Comment:
I can see only one `ip_configuration` block in this firewall resource, and the example name is `azfw-multi-addresses`, is this correct config?

Reply:
good catch, thanks, added an extra ipconfig